src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -30,6 +30,45 @@ internal void VerifyDiagnostics(string source, CSharpCompilationOptions options,
     public class LocalFunctionTests : LocalFunctionsTestBase
     {
         [Fact]
+        public void TypeParameterBindingScope()
+        {
+            var src = @"
+class C
+{
+    public void M()
+    {
+        {
+            int T = 0; // Should not have error
+
+            int Local<T>() => 0; // Should conflict with above
```

Comment:
>Local [](start = 16, length = 5)

It would be good to have a test when the name of the local function conflict with the name of containing function/local function, or with its type parameter.